mongo/src/main/scala/it/agilelab/darwin/connector/mongo/MongoConnector.scala

```diff
@@ -1,19 +1,129 @@
 package it.agilelab.darwin.connector.mongo
 
-import com.typesafe.config.Config
-import it.agilelab.darwin.common.Connector
+import com.mongodb.{BasicDBObject, ErrorCategory}
+import it.agilelab.darwin.common.{Connector, Logging}
+import it.agilelab.darwin.connector.mongo.ConfigurationMongoModels.BaseMongoConfig
 import org.apache.avro.Schema
+import org.apache.avro.Schema.Parser
+import org.mongodb.scala.bson.BsonDocument
+import org.mongodb.scala.{Document, MongoClient, MongoCollection, MongoWriteException, bson}
 
-class MongoConnector(config: Config) extends Connector {
-  override def fullLoad(): Seq[(Long, Schema)] = ???
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Await
+import scala.util.{Failure, Success}
 
-  override def insert(schemas: Seq[(Long, Schema)]): Unit = ???
+class MongoConnector(mongoClient: MongoClient, mongoConfig: BaseMongoConfig) extends Connector with Logging {
 
-  override def createTable(): Unit = ???
+  private def parser: Parser = new Parser()
 
-  override def tableExists(): Boolean = ???
+  override def fullLoad(): Seq[(Long, Schema)] = {
 
-  override def tableCreationHint(): String = ???
+    log.debug(s"loading all schemas from collection ${mongoConfig.collection}")
+    val collection =
+      mongoClient
+        .getDatabase(mongoConfig.database)
+        .getCollection(mongoConfig.collection)
 
-  override def findSchema(id: Long): Option[Schema] = ???
+    val schemas: Seq[Option[(Long, Schema)]] =
+      Await.result(
+        collection.find().map(document => {
+          try {
+            val key = document.filterKeys(k => k == "_id").head._2.asInt64().getValue
+            val schema = parser.parse(document.filterKeys(k => k == "schema").head._2.asString().getValue)
+            Some(key -> schema)
+          } catch {
+            case exception: Exception =>
+              log.info(exception.getMessage)
+              None
+          }
+        }).toFuture(),
+        mongoConfig.timeout
+      )
+    log.debug(s"${schemas.size} loaded from MongoDB")
+    schemas.flatten
```

Comment:
The other connectors fails if any schema is not parsable, I would keep the semantic homogeneous (it's debatable if it's a good strategy or not, but I would like to keep it the same for all the connectors).
I propose the following implementation, tell me what you think about it, and feel free to accept the suggestion if it's ok for you:
```suggestion
    // this method returns specific errors on what happened and wraps "inner" exceptions if present
    def extract[A](d: Document, fieldName: String, f: BsonValue => A): Try[A] = {
      d.filterKeys(k => k == fieldName)
        .headOption
        .fold[Try[A]](Failure(new RuntimeException(s"Cannot find $fieldName field in document"))) {
          case (_, value) =>
            Try(f(value)).recoverWith {
              case t: Throwable =>
                Failure(new RuntimeException(s"$fieldName was not of expected type", t))
            }
        }
    }

    val schemas: Seq[Try[(Long, Schema)]] =
      Await.result(
        collection.find().map { document =>
          for {
            key <- extract(document, "_id", _.asInt64().getValue)
            schemaStr <- extract(document, "schema", _.asString().getValue)
            schema <- Try(parser.parse(schemaStr))
          } yield key -> schema
        }.toFuture(),
        mongoConfig.timeout
      )
    log.debug(s"${schemas.size} loaded from MongoDB")
    // this way the first exception is thrown, but we can change this line to support different error handling strategies
    schemas.map(_.get) 
```